[Op][Refactor] Decl make-functions in local headers. Namespace alignment cont.

MasterJH5574 opened the pull request:
Following #81, the main focus of this PR is to expose the make-functions of every operators into local headers. And meanwhile remove the prepended `"Make"` prefix of the make-function names, so that whenever people want to call certain operator, they can make it by include the corresponding header and invoke the make-functions, without constructing the attribute and the op call manually.

Besides, this PR continues to align the name/namespace of the operators so far, according to [DataAPI](https://data-apis.org/array-api/latest/API_specification/index.html). Specifically, this PR
* move Matmul from NN namespace to “linear algebra” namespace;
* rename Variance to Var.

Several to-do items or undetermined items:
- [x] Do the make-function refactoring for the builtin operators, memory operators, etc.
- [x] Update the use sites of operator calls.
- [x] Rename Cast to Astype per DataAPI (?) https://data-apis.org/array-api/latest/API_specification/generated/array_api.astype.html#array_api.astype

cc @tqchen

MasterJH5574 committed:
Align op namespace with DataAPI (Matmul, Cast -> Astype)

Separating Matmul into "linear algebra" namespace (with "Var" reverted)

Astype

MasterJH5574 commented:
Well it seems that changing “variance” to “var” will come with conflict, as “var” is usually understood as “variable.” I’ll change it back with proper notes.

MasterJH5574 committed:
[Op] Decl Make-functions in local headers

tqchen commented on src/relax/op/tensor/statistical.h:
```diff
@@ -0,0 +1,73 @@
+  }                                                                      \
+  TVM_REGISTER_GLOBAL("relax.op." OpRegName).set_body_typed(OpName)
+
+Expr Sum(Expr data, Optional<Array<Integer>> axis, bool keepdims);
```
naming, use low-level case Sum=>sum to be consistent with data api

MasterJH5574 replied:
If so then many operators need be renamed to lowercase. Will do later.

tqchen commented on src/relax/op/nn/nn.h:
```diff
@@ -0,0 +1,58 @@
+namespace tvm {
+namespace relax {
+
+Expr ReLU(Expr e);
```
add some docstring given it is in header

MasterJH5574 committed:
Op rename to lowercase, header docstring, macro enhancement

MasterJH5574 commented:
@tqchen Some notes on the latest commit:
* Renamed all op to lowercase on C++ side, including those in DataAPI and those not in, basically to keep consistency.
* Added docstring to the op decl in headers. Not too much docstring content for NN ops and image ops though. They are well documented on Python side.
* Enhanced the macros for binary and statistical ops.
